src/Aspire.Hosting.SqlServer/SqlServerBuilderExtensions.cs

```diff
@@ -112,4 +162,87 @@ public static IResourceBuilder<SqlServerServerResource> WithDataBindMount(this I
+
+        builder.WithAnnotation(new CreationScriptAnnotation(script));
+
+        return builder;
+    }
+
+    private static async Task CreateDatabaseAsync(SqlConnection sqlConnection, SqlServerDatabaseResource sqlDatabase, IServiceProvider serviceProvider, CancellationToken ct)
+    {
+        try
+        {
+            var scriptAnnotation = sqlDatabase.Annotations.OfType<CreationScriptAnnotation>().LastOrDefault();
+
+            if (scriptAnnotation?.Script == null)
+            {
+                var quotedDatabaseIdentifier = new SqlCommandBuilder().QuoteIdentifier(sqlDatabase.DatabaseName);
+                using var command = sqlConnection.CreateCommand();
+                command.CommandText = $"IF ( NOT EXISTS ( SELECT 1 FROM sys.databases WHERE name = @DatabaseName ) ) CREATE DATABASE {quotedDatabaseIdentifier};";
+                command.Parameters.Add(new SqlParameter("@DatabaseName", sqlDatabase.DatabaseName));
+                await command.ExecuteNonQueryAsync(ct).ConfigureAwait(false);
+            }
+            else
+            {
+                using var reader = new StringReader(scriptAnnotation.Script);
+                var batchBuilder = new StringBuilder();
+
+                while (reader.ReadLine() is { } line)
+                {
+                    var matchGo = GoStatements().Match(line);
+
+                    if (matchGo.Success)
+                    {
+                        // Execute the current batch
+                        var count = matchGo.Groups["repeat"].Success ? int.Parse(matchGo.Groups["repeat"].Value, CultureInfo.InvariantCulture) : 1;
+                        var batch = batchBuilder.ToString();
+
+                        for (var i = 0; i < count; i++)
+                        {
+                            using var command = sqlConnection.CreateCommand();
+                            command.CommandText = batch;
+                            await command.ExecuteNonQueryAsync(ct).ConfigureAwait(false);
+                        }
+
+                        batchBuilder.Clear();
+                    }
+                    else
+                    {
+                        // Prevent batches with only whitespace
+                        if (!string.IsNullOrWhiteSpace(line))
+                        {
+                            batchBuilder.AppendLine(line);
+                        }
+                    }
+                }
+
+                // Process the remaining batch lines
+                if (batchBuilder.Length > 0)
+                {
+                    using var command = sqlConnection.CreateCommand();
+                    command.CommandText = batchBuilder.ToString();
+                    await command.ExecuteNonQueryAsync(ct).ConfigureAwait(false);
+                }
+            }
+        }
+        catch (Exception e)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<DistributedApplicationBuilder>>();
```

Comment:
Are we going to update this to write the log to the resource instead?

Reply:
I just got taught how to, yes I will change that right away.